src/test/java/org/jenkinsci/plugins/workflow/properties/PipelineTriggersJobPropertyTest.java

```diff
@@ -0,0 +1,182 @@
+        Boolean currentStatus = t.currentStatus();
+        assertNotNull(currentStatus);
+        assertTrue(currentStatus);
+    }
+
+    @Test
+    public void previousTriggerStopped() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        MockTrigger t = new MockTrigger();
+        p.addTrigger(t);
+        p.save();
+        MockTrigger t2 = new MockTrigger();
+        p.addTrigger(t2);
+        p.save();
+
+        assertFalse(t.isStarted);
+        assertTrue(t2.isStarted);
+    }
+
+    @LocalData
+    @Test
+    public void triggerMigration() throws Exception {
+        WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+        assertNotNull(p);
+
+        PipelineTriggersJobProperty triggerProp = p.getProperty(PipelineTriggersJobProperty.class);
+        assertNotNull(triggerProp);
+
+        assertEquals(2, triggerProp.getTriggers().size());
+        assertEquals(2, p.getTriggers().size());
+
+        Trigger timerFromProp = getTriggerFromList(TimerTrigger.class, triggerProp.getTriggers());
+        assertNotNull(timerFromProp);
+        assertEquals(TimerTrigger.class, timerFromProp.getClass());
+
+        Trigger timerFromJob = p.getTriggers().get(timerFromProp.getDescriptor());
+        assertEquals(timerFromProp, timerFromJob);
+
+        Trigger mockFromProp = getTriggerFromList(MockTrigger.class, triggerProp.getTriggers());
+        assertNotNull(mockFromProp);
+        assertEquals(MockTrigger.class, mockFromProp.getClass());
+
+        Trigger mockFromJob = p.getTriggers().get(mockFromProp.getDescriptor());
+        assertEquals(mockFromProp, mockFromJob);
+
+        assertNotNull(((MockTrigger)mockFromProp).currentStatus());
+        assertEquals("[null, false, null, false, null, false]", MockTrigger.startsAndStops.toString());
+    }
+
+    @Test
+    public void configRoundTrip() throws Exception {
+        WorkflowJob defaultCase = r.jenkins.createProject(WorkflowJob.class, "defaultCase");
+        assertTrue(defaultCase.getTriggers().isEmpty());
+
+        WorkflowJob roundTripDefault = r.configRoundtrip(defaultCase);
+        assertTrue(roundTripDefault.getTriggers().isEmpty());
+
+        WorkflowJob withTriggerCase = r.jenkins.createProject(WorkflowJob.class, "withTriggerCase");
+        withTriggerCase.addTrigger(new MockTrigger());
+        assertEquals(1, withTriggerCase.getTriggers().size());
+        List<Trigger<?>> origTriggers = new ArrayList<>(withTriggerCase.getTriggers().values());
+
+        assertEquals(MockTrigger.class, origTriggers.get(0).getClass());
+
+        WorkflowJob roundTripWithTrigger = r.configRoundtrip(withTriggerCase);
+
+        assertEquals(1, roundTripWithTrigger.getTriggers().size());
+        List<Trigger<?>> modTriggers = new ArrayList<>(roundTripWithTrigger.getTriggers().values());
+
+        assertEquals(MockTrigger.class, modTriggers.get(0).getClass());
```

Comment:
:ant: would be nice to include some trigger with configuration, such as `TimerTrigger`, in the list and verify that this configuration got correctly round-tripped by `f:descriptorList`.